nomad/structs/structs.go

```diff
@@ -11403,8 +11419,9 @@ type PlanResult struct {
 
 // IsNoOp checks if this plan result would do nothing
 func (p *PlanResult) IsNoOp() bool {
-	return len(p.NodeUpdate) == 0 && len(p.NodeAllocation) == 0 &&
-		len(p.DeploymentUpdates) == 0 && p.Deployment == nil
+	return len(p.IneligibleNodes) == 0 && len(p.NodeUpdate) == 0 &&
```

Comment:
This new check will cause the [no-op fast-path](https://github.com/hashicorp/nomad/blob/v1.3.1/nomad/plan_apply.go#L147-L151) to be skipped in cases there it previously wouldn't, so it may be worth double checking if nothing will break.